monkey/infection_monkey/monkey.py

```diff
@@ -215,6 +218,15 @@ def _discover_os(self):
         )
         self._agent_event_queue.publish(event)
 
+    def _discover_hostname(self):
+        event = HostnameDiscoveryEvent(
+            source=self._agent_id,
+            timestamp=time.time(),
+            tags={T1592_ATTACK_TECHNIQUE_TAG},
```

Comment:
I don't think [1592](https://attack.mitre.org/techniques/T1592/) is the correct attack technique (for OS discovery, either). It's for collecting information about remote systems, not local systems.

[T1082](https://attack.mitre.org/techniques/T1082/) is a better choice for both hostname and OS discovery.

Reply:
[1592](https://attack.mitre.org/techniques/T1592/) is not part of PingScanEvent though 🤔